internal/document/frontmatter.go

```diff
@@ -67,10 +87,104 @@ func ParseFrontmatter(raw string) (f Frontmatter, info FrontmatterParseInfo) {
 	return
 }
 
+func ToJSONStr(dest map[string]interface{}, bytes []byte) string {
+	if err := json.Unmarshal(bytes, &dest); err != nil {
+		return ""
+	}
+
+	bytes, _ = json.Marshal(dest)
+	return fmt.Sprintf("---\n%s\n---", string(bytes))
+}
+
+func ToYamlStr(dest map[string]interface{}, bytes []byte) string {
+	if err := yaml.Unmarshal(bytes, &dest); err != nil {
+		return ""
+	}
+
+	var buf byteslib.Buffer
+	encoder := yaml.NewEncoder(&buf)
+	encoder.SetIndent(2)
+	_ = encoder.Encode(&dest)
+	bytes = buf.Bytes()
+	_ = encoder.Close()
+
+	return fmt.Sprintf("---\n%s---", string(bytes))
+}
+
+func ToTomlStr(dest map[string]interface{}, bytes []byte) string {
+	if err := toml.Unmarshal(bytes, &dest); err != nil {
+		return ""
+	}
+
+	bytes, _ = toml.Marshal(dest)
+	return fmt.Sprintf("+++\n%s+++", string(bytes))
+}
+
+// ReEncodeFrontmatter converts Frontmatter to a string based on the provided format.
+func ReEncodeFrontmatter(raw string, f Frontmatter, info FrontmatterParseInfo) string {
+	f.EnsureID()
+
+	fmMap := make(map[string]interface{})
+	fmMap["runme"] = f.Runme
+
+	bytes := []byte(raw)
+	lines := strings.Split(raw, "\n")
+
+	if len(bytes) == 0 {
+		return ToYamlStr(fmMap, bytes)
+	}
+
+	if len(lines) < 2 || strings.TrimSpace(lines[0]) != strings.TrimSpace(lines[len(lines)-1]) {
+		info.other = errors.New("invalid frontmatter 2")
+		return ""
+	}
+
+	raw = strings.Join(lines[1:len(lines)-1], "\n")
+	bytes = []byte(raw)
+
+	switch {
+	case info.yaml == nil:
+		return ToYamlStr(fmMap, bytes)
+	case info.json == nil:
+		return ToJSONStr(fmMap, bytes)
+	case info.toml == nil:
+		return ToTomlStr(fmMap, bytes)
+	default:
+		return ToYamlStr(fmMap, bytes)
+	}
+}
+
+func (fmtr *Frontmatter) EnsureID() {
+	if !idgen.ValidID(fmtr.Runme.ID) {
+		fmtr.Runme.ID = idgen.GenerateID()
+	}
+
+	if fmtr.Runme.Version == "" {
+		fmtr.Runme.Version = version.BuildVersion
+	}
+}
+
 func (fmtr Frontmatter) ToParser() *parserv1.Frontmatter {
 	return &parserv1.Frontmatter{
+		Runme: &parserv1.Runme{
+			Id:      fmtr.Runme.ID,
+			Version: fmtr.Runme.Version,
+		},
 		Shell:       fmtr.Shell,
 		Cwd:         fmtr.Cwd,
 		SkipPrompts: fmtr.SkipPrompts,
 	}
 }
+
+// InjectYamlTestID injects a test id into a yaml document
+func InjectYamlTestID(s string) string {
```

Comment:
I'm not sure if this is a correct description. It seems like `InjectFrontmatter(documentString, runmeID, runmeVersion string) string` might explain the purpose better. It may also make more sense to place it in `internal/document/editor`.